### DISCUSSION TITLE
Out of memory for 3GB .pbf file?

### DISCUSSION THREAD

**Discussion by Afler** [mediagis/nominatim-docker#405] [Q&A] [answered]
I've tried to run the following .bat file on Windows 10 with 8GB RAM 32GB SSD allocated for Docker:

```
docker run -it --shm-size="8g" ^
-e PBF_URL=https://download.geofabrik.de/russia-latest.osm.pbf ^
-e REPLICATION_URL=https://download.geofabrik.de/russia-updates/ ^
-e POSTGRES_MAINTENANCE_WORK_MEM=5GB ^
-e POSTGRES_EFFECTIVE_CACHE_SIZE=12GB ^
-p 8085:8080 ^
--name nominatim ^
mediagis/nominatim:4.2
```

russia-latest.osm.pbf is 3.2 GB.
I've had the following docker resource settings:
![image](https://user-images.githubusercontent.com/49097219/207301395-2867523b-0033-4b28-b53c-789b83a27840.png)
And eventually I've ran into the following error:

```
2022-12-13 10:51:51  Parsing gazetteer style file '/usr/local/etc/nominatim/import-full.style'.
NOTICE:  table "place" does not exist, skipping
Processing: Node(478820k 1057.0k/s) Way(0k 0.00k/s) Relation(0 0.0/s)
2022-12-13 10:59:24  ERROR: DB copy thread failed: Ending COPY mode for 'planet_osm_nodes' failed: ERROR:  could not extend file "base/16386/19931.16": No space left on device
HINT:  Check free disk space.
CONTEXT:  COPY planet_osm_nodes, line 388297000
```

I don't clearly understand how i can ran out of disk space with 3.2GB .pbf file with allocated 32GB of disk space and how can i deal with it?

**Comment by leonardehrenfried** [comment, marked as answer]
Nominatim needs a lot of disk space. The pbf is a highly optimized binary format and Nominatim stores everything in Postgres which is optimized for fast reading but not disk space.

**Reply by Afler** [reply]
Is there a way to evaluate a memory top limit i need to successfully install certain country/region in lack of memory resources case?

**Reply by leonardehrenfried** [reply]
I think you will have to use trial and error.

**Reply by Afler** [reply]
TY for the answer!

**Reply by mausch** [reply]
Just for reference, when I imported Greater London (77MB file) it needed 1.7GB memory.
Importing all of Europe required 23GB memory.
This was about 2 years ago.
